src/meta/src/hummock/manager/mod.rs

```diff
@@ -493,3 +501,68 @@ async fn write_exclusive_cluster_id(
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct TableWriteThroughputStatistic {
+    pub throughput: u64,
+    pub timestamp: i64,
+}
+
+#[derive(Debug, Clone)]
+pub struct TableWriteThroughputStatisticManager {
+    table_throughput: HashMap<u32, Vec<TableWriteThroughputStatistic>>,
+    max_statistic_expired_time: i64,
+}
+
+impl TableWriteThroughputStatisticManager {
+    pub fn new(max_statistic_expired_time: i64) -> Self {
+        Self {
+            table_throughput: HashMap::new(),
+            max_statistic_expired_time,
+        }
+    }
+
+    pub fn add_table_throughput_with_ts(&mut self, table_id: u32, throughput: u64, timestamp: i64) {
+        let table_throughput = self
+            .table_throughput
+            .entry(table_id)
+            .or_insert_with(Vec::new);
+        table_throughput.push(TableWriteThroughputStatistic {
+            throughput,
+            timestamp,
+        });
+
+        table_throughput
+            .retain(|statistic| timestamp - statistic.timestamp <= self.max_statistic_expired_time);
+    }
+
+    pub fn get_table_throughput(
+        &self,
+        table_id: u32,
+        expired_time: i64,
+    ) -> Option<Vec<&TableWriteThroughputStatistic>> {
+        let timestamp = chrono::Utc::now().timestamp();
```

Comment:
minor: it is better to put the unit of the timestamp in the name: `timestamp_secs`